content/techniques/versioning.md

```diff
@@ -87,6 +91,48 @@ The `key` property should be the key and separator of the key-value pair that co
 
 > info **Hint** The `VersioningType` enum is available to use for the `type` property and is imported from the `@nestjs/common` package.
 
+#### Custom Versioning Type
+Custom Versioning uses any aspect of the request to specify the version (or versions). The incoming request is analyzed
+using an `extractor` function that returns a string or array of strings.
+
+If multiple versions are provided by the requester, the extractor function can return an array of strings, sorted in
+order of greatest/highest version to smallest/lowest version. Versions are matched to routes in order from highest to
+lowest.
+
+If an empty string or array is returned from the `extractor`, no routes are matched and a 404 is returned.
+
+For example, if an incoming request specifies it supports versions `1`, `2`, and `3`, the `extractor` **MUST** return `[3,
+2, 1]`. This ensures that the highest possible route version is selected first.
+
+If versions `[3, 2, 1]` are extracted, but routes only exist for version `2` and `1`, the route that matches version `2`
+is selected (version `3` is automatically ignored).
+
+> warning **Notice** Selecting the highest matched version based on the array returned from `extractor`
+> **does not reliably work** with the Express adapter due to design limitations. A single version (either a string or
+> array of 1 element) works just fine in Express. Fastify correctly supports both highest matched version
+> selection and single version selection.
+
+To enable **Custom Versioning** for your application, create an `extractor` function and pass it into your application
+like so:
+```typescript
+@@filename(main)
+// Example extractor that pulls out a list of versions from a custom header and turns it into a sorted array.
+// This example uses Fastify, but Express requests can be processed in a similar way.
```

Comment:
I think this example is not really fitting.. Because if you're just reading the versions from a custom header field, you could also use the "custom header field" approach..

Reply:
Not sure what you're recommending here. Sure there's an existing Header versioning method, as well as an existing Media Type versioning method, both of which use headers, and both of which do not offer enough adaptability to address this use case (which yes, uses a header).

The Header versioning method doesn't work for `Accept` headers, and the Media Type versioning method is limited to the version key being in a very specific part of the `Accept` header value.

If anything, this example shows how to address the gap provided by those two versioning methods.